test/release/examples/primers/arrays.chpl

```diff
@@ -339,20 +351,35 @@ VarDom = {1..n};
 writeln("VarArr should now be reset: ", VarArr, "\n");
 
 //
-// Note that querying an array's domain via the ``.domain`` method or
-// the function argument query syntax does not result in a domain
-// expression that can be reassigned.  In particular, we cannot do:
+// This trick can be used to create records or classes with array fields which
+// may have different sizes in each instance of the type:
 //
-//   ``VarArr.domain = {1..2*n};``
+
+record R {
+  var FieldDom = {1..0};
+  var FieldArr: [FieldDom] int;
+
+  proc init(size) {
+    FieldDom = {1..size};
+  }
+}
+
+var RSmall = new R(10);
+var RLarge = new R(1000);
+
+writeln("Size of RSmall's FieldArr: ", RSmall.FieldArr.size);
+writeln("Size of RLarge's FieldArr: ", RLarge.FieldArr.size);
+
 //
-// nor:
+// Note: further information on records can be found in the
+// :ref:`Records Primer <primers-records>`
 //
-//   ``proc foo(X: [?D]) {  D = {1..2*n};  }``
+
 //
 // Only a domain variable or formal argument can be
 // reassigned to reallocate arrays.  This is to avoid confusion
-// since assigning one domain variable can cause a number of
-// arrays to be reallocated.  It also implies that arrays declared
+// since assigning a domain using ``.domain`` query on one array can cause a number of
+// other arrays to be reallocated.  It also implies that arrays declared
```

Comment:
I feel as though this content should likely be integrated into the earlier section you added about `.domain` or `?D` style queries.  Specifically, it feels as though this is giving the rationale, which either should've been given up there rather than being separated from it by the other content, or potentially it could even be dropped.

Pre-existing condition, but if this text persists in some way, "formal argument" should probably become "`ref` argument"?

Reply:
> I feel as though this content should likely be integrated into the earlier section you added about .domain or ?D style queries. Specifically, it feels as though this is giving the rationale, which either should've been given up there rather than being separated from it by the other content, or potentially it could even be dropped.

Makes sense. Though, between there and here, most of the discussion is around reallocating arrays through domain reassignment. Maybe I can reword it a little bit better to say "remember what I said above `.domain` being `const ref`?"

Reply:
New big thought here (sorry that I didn't spend the time earlier to have it then... I was racing to finish before the morning's meetings):

When I scan the file on your branch as a whole, it feels like we now talk about how "You can't assign to a domain via `.domain` or `?D`" long before we even define what assigning a domain does, which seems like it will lead to questions like "Wait, why would I even want to do that?"  So now I'm thinking that your first set of edits should be moved way down to where we do define array reallocation through domain assignment, say, maybe right before the legacy text about "One trick..."?

> Though, between there and here, most of the discussion is around reallocating arrays through domain reassignment. 

True, though I feel like this doesn't add much that's new above what you've done above.  Taking it sentence by sentence to see if I believe that with more thought:

> // Only a domain variable or formal argument can be                             
> // reassigned to reallocate arrays.

This seems similar to what I was suggesting inserting after your original paragraph to switch from "This won't work" to "This is what will."  Maybe by moving that content down here, you won't need that text or this text anymore?

>  // This is to avoid confusion                 
> // since assigning one domain variable can cause a number of                    
> // arrays to be reallocated.

This is interesting, but doesn't seem essential if we're interested in trimming fat (e.g., rationales could be relegated to the language spec rather than the primer).  I'm not opposed to keeping it either if you like it / it flows.

> //  It also implies that arrays declared              
> // using an anonymous domain cannot be reallocated.   So for our                 
> // original array declarations ``A`` and ``B``, we have no way of reallocating  
> // them.  Arrays with constant domains provide the compiler with                
> // optimization benefits, so this supports a common case efficiently.           

This seems more interesting and more closely related to your first set of edits (that I'm proposing moving down here).